### PR TITLE
sort-students

### PR DESCRIPTION
**Pull request by DenisGurskiy**

### CODE REVIEW

File: src/sortStudents.ts [hunk 1]
```diff
@@ -1,16 +1,96 @@
...
+
+      case SortType.Surname:
+        if (order === 'asc') {
+          return a.surname.localeCompare(b.surname);
+        }
+
+        if (order === 'desc') {
+          return b.surname.localeCompare(a.surname);
+        }
+        break;
```
Comment (VitaliyBondarenko1982) [inline src/sortStudents.ts:66]:
```suggestion
      case SortType.Name:
      case SortType.Surname:
        return order === 'asc' 
          ? a[sortBy].localeCompare(b[sortBy]) 
           : b[sortBy].localeCompare(a[sortBy);
        break;
```
unite cases with same logic

File: src/sortStudents.ts [hunk 2]
```diff
@@ -1,16 +1,96 @@
...
+
+      case SortType.Married:
+        if (order === 'asc') {
+          return Number(a.married) - Number(b.married);
+        }
+
+        if (order === 'desc') {
+          return Number(b.married) - Number(a.married);
+        }
+        break;
```
Comment (VitaliyBondarenko1982) [inline src/sortStudents.ts:86]: for Age and Married is same logic. 
```suggestion
      case SortType.Age:
      case SortType.Married:
         return order === 'asc' 
          ? Number(a.married) - Number(b.married)
           : Number(b.married) - Number(a.married);

        break;
```

File: src/sortStudents.ts [hunk 3]
```diff
@@ -1,16 +1,96 @@
...
-export function sortStudents(students, sortBy, order) {
-  // write your function
+  resultStudents.sort((a: Student, b: Student): number => {
+    switch (sortBy) {
+      case SortType.AverageGrade:
+        averageA = a.grades.reduce((sum: number, x: number) => sum + x)
+          / a.grades.length;
+
+        averageB = b.grades.reduce((sum: number, x: number) => sum + x)
+          / b.grades.length;
```
Comment (VitaliyBondarenko1982) [inline src/sortStudents.ts:37]: consider to make helper function for counting average.

**Review by VitaliyBondarenko1982 — CHANGES_REQUESTED** [review]

File: src/sortStudents.ts (hunk 1 above)
Comment (DenisGurskiy) [reply src/sortStudents.ts:66]: The sortBy variable contains the values ​​of SortType.Name and SortType.Surname. How to get to the Name and Surname properties of objects through this variable?

File: src/sortStudents.ts (hunk 2 above)
Comment (DenisGurskiy) [reply src/sortStudents.ts:86]: The same question. Writing a[sortBy] throws an error.

**Review by volodymyr-soltys97 — APPROVED** [review]
Nice work 🔥

**Pull request closed without merging**